src/main/java/io/github/fvarrui/javapackager/maven/CreateWindowsExeLaunch4j.java

```diff
@@ -89,6 +90,17 @@ protected File doApply(WindowsPackager packager) throws Exception {
 						element("addDependencies", "false")
 					)
 				);
+		
+		SingleInstance singleInstance = winConfig.getSingleInstance();
+		if(singleInstance != null){
+			pluginConfig.add(
+					element("singleInstance",
+							element("mutexName", singleInstance.getMutexName()),
+							element("windowTitle", singleInstance.getWindowTitle())
```

Comment:
Are both parameters mandatory? What if one of them is not specified?

Reply:
I'm glad to provide some help. Here are the documents related to launch4j:

[https://launch4j.sourceforge.net/docs.html](https://launch4j.sourceforge.net/docs.html)
```
<singleInstance>
  Optional, allow to run only a single instance of the application.
  <mutexName>
    Unique mutex name that will identify the application.
  <windowTitle>
    Optional, recognized by GUI header only. Title or title part of a window to bring up instead of running a new instance.
```

I conducted some tests on the Windows system and the results are as follows:

- \<mutexName> is required, if not set, it cannot be a single instance

- \<windowTitle> is optional. If it is not set or the value set is different from the application window title, the program can maintain a single instance, but it cannot be displayed to the frontend when the program is minimized